### PR TITLE
Fix error in verifying same hash

### PR DESCRIPTION
**Pull request by afcidk**
`hash_chain_verify` should return success if given the same hash.

### CODE REVIEW

File: hashchain.c [hunk 1]
```diff
@@ -76,6 +76,9 @@ bool hash_chain_verify(const void *h,
     EVP_MD_CTX *ctx;
     int result;
     int digest_len = EVP_MD_size(hash);
+    if (memcmp(h, tip, digest_len) == 0)
+        return 1;
```
Comment (jserv) [inline hashchain.c]: By the way, it should be `true` or `false` since its return type is specific to `bool`. Please send another pull request to rework.

Comment (afcidk) [reply hashchain.c]: Fixed.
